idaes/models_extra/power_generation/unit_models/soc_submodels/channel.py

```diff
@@ -0,0 +1,839 @@
+                initialize=0,
+                units=pyo.units.mol / pyo.units.m**3,
+            )
+            self.material_flux_x0 = pyo.Var(
+                tset,
+                iznodes,
+                comps,
+                doc="Material flux from electrode surface to channel "
+                "(positive is in)",
+                initialize=0,
+                units=pyo.units.mol / pyo.units.m**2 / pyo.units.s,
+            )
+
+            @self.Expression(tset, iznodes, comps)
+            def conc_mol_comp_deviation_x1(b, t, iz, j):
+                return 0
+
+            @self.Expression(tset, iznodes, comps)
+            def material_flux_x1(b, t, iz, j):
+                return 0
+
+        else:
+            self.conc_mol_comp_deviation_x1 = pyo.Var(
+                tset,
+                iznodes,
+                comps,
+                doc="Deviation of concentration at electrode surface from that "
+                "in the bulk channel",
+                initialize=0,
+                units=pyo.units.mol / pyo.units.m**3,
+            )
+            self.material_flux_x1 = pyo.Var(
+                tset,
+                iznodes,
+                comps,
+                doc="Material flux from channel to electrode surface "
+                "(positive is out)",
+                initialize=0,
+                units=pyo.units.mol / pyo.units.m**2 / pyo.units.s,
+            )
+
+            @self.Expression(tset, iznodes, comps)
+            def conc_mol_comp_deviation_x0(b, t, iz, j):
+                return 0
+
+            @self.Expression(tset, iznodes, comps)
+            def material_flux_x0(b, t, iz, j):
+                return 0
+
+        # Channel thickness AKA length in the x direction is specific to the
+        # channel so local variable here is the only option
+        self.length_x = pyo.Var(
+            doc="Thickness from interconnect to electrode (x-direction)",
+            units=pyo.units.m,
+        )
+        self.heat_transfer_coefficient = pyo.Var(
+            tset,
+            iznodes,
+            doc="Local channel heat transfer coefficient",
+            initialize=500,
+            units=pyo.units.J / pyo.units.m**2 / pyo.units.s / pyo.units.K,
+        )
+        self.flow_mol = pyo.Var(
+            tset,
+            iznodes,
+            doc="Molar flow in the z-direction through faces",
+            units=pyo.units.mol / pyo.units.s,
+            bounds=(-1e-9, None),
+        )
+        self.conc_mol_comp = pyo.Var(
+            tset,
+            iznodes,
+            comps,
+            doc="Component concentration at node centers",
+            units=pyo.units.mol / pyo.units.m**3,
+            bounds=(0, None),
+        )
+        self.temperature_deviation_x = pyo.Var(
+            tset,
+            iznodes,
+            doc="Deviation of temperature at node centers " "from temperature_z",
+            units=pyo.units.K,
+            bounds=(-1000, 1000),
```

Comment:
Yeah, I expect them to mostly be on the order of 10 Kelvin, but don't want to assume that a priori. The reason for temperature bounds was to prevent IPOPT from taking enormous steps, then getting AMPL errors when the (absolute) temperature is negative. If it violates a bound, it still does a line search, but it doesn't fill your console with output.